emit file lost import module in .js with package use export = with build by new typescript@next

<!-- 🚨 STOP 🚨 STOP 🚨 STOP 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!--
Please try to reproduce the issue with the latest published version. It may have already been fixed.
For npm: `typescript@next`
This is also the 'Nightly' version in the playground: http://www.typescriptlang.org/play/?ts=Nightly
-->
**TypeScript Version:**  3.7.x-dev.201xxxxx

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:**

**Code**

Version 4.0.0-dev.20200615

[bug-web-46035.zip](https://github.com/microsoft/TypeScript/files/4800087/bug-web-46035.zip)

download the zip and run `yarn build` will see

**note: remember run yarn for install deps before test**

```ts

import gitRoot from 'git-root2';

let git_root = gitRoot(process.cwd())

console.dir(git_root)


```

**Expected behavior:**

https://www.typescriptlang.org/v2/en/play?module=1#code/JYWwDg9gTgLgBAc2DAShC8BmUIjgciRgFocMAmfAbgCgaAbAU3iIH0z4BeRZNDACjA4AxowDOYgHTCA7gBN+ASkV1hEAHZiITSXOBR+bDoqA

```js
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const git_root2_1 = __importDefault(require("git-root2"));
let git_root = git_root2_1.default(process.cwd());
console.dir(git_root);
```

**Actual behavior:**

```js

Object.defineProperty(exports, "__esModule", { value: true });
let git_root = gitRoot(process.cwd());
console.dir(git_root);
//# sourceMappingURL=index.js.map
```

**Playground Link:** <!-- A link to a TypeScript Playground "Share" link which demonstrates this behavior -->

**Related Issues:** <!-- Did you find other bugs that looked similar? -->

Interestingly, when running the following:

```
yarn
yarn build
```

I get the following error:
```
index.ts:7:16 - error TS2552: Cannot find name 'gitRoot'. Did you mean 'git_root'?

7 let git_root = gitRoot(process.cwd())
                 ~~~~~~~

  index.ts:7:5
    7 let git_root = gitRoot(process.cwd())
          ~~~~~~~~
    'git_root' is declared here.


Found 1 error.
```

However, if I change the source file to this:

```ts
import { default as gitRoot } from "git-root2";
...
```

We both check and emit correctly (i.e., including `__importDefault`).

For some reason we aren't treating the default import as a valid identifier reference.

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -2497,10 +2497,7 @@ namespace ts {
 
         function resolveExportByName(moduleSymbol: Symbol, name: __String, sourceNode: TypeOnlyCompatibleAliasDeclaration | undefined, dontResolveAlias: boolean) {
             const exportValue = moduleSymbol.exports!.get(InternalSymbolName.ExportEquals);
-            if (exportValue) {
-                return getPropertyOfType(getTypeOfSymbol(exportValue), name);
-            }
-            const exportSymbol = moduleSymbol.exports!.get(name);
+            const exportSymbol = exportValue ? getPropertyOfType(getTypeOfSymbol(exportValue), name) : moduleSymbol.exports!.get(name);
             const resolved = resolveSymbol(exportSymbol, dontResolveAlias);
             markSymbolOfAliasDeclarationIfTypeOnly(sourceNode, exportSymbol, resolved, /*overwriteEmpty*/ false);
             return resolved;

--- a/tests/baselines/reference/exportAssignmentOfExportNamespaceWithDefault.js
+++ b/tests/baselines/reference/exportAssignmentOfExportNamespaceWithDefault.js
@@ -1,0 +1,31 @@
+//// [tests/cases/conformance/externalModules/exportAssignmentOfExportNamespaceWithDefault.ts] ////
+
+//// [main.ts]
+// https://github.com/microsoft/TypeScript/issues/39149
+import a from "a";
+a();
+
+//// [external.d.ts]
+declare module "b" {
+    export function a(): void;
+    export namespace a {
+        var _a: typeof a;
+        export { _a as default };
+    }
+    export default a;
+}
+
+declare module "a" {
+    import { a } from "b";
+    export = a;
+}
+
+//// [main.js]
+"use strict";
+var __importDefault = (this && this.__importDefault) || function (mod) {
+    return (mod && mod.__esModule) ? mod : { "default": mod };
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+// https://github.com/microsoft/TypeScript/issues/39149
+const a_1 = __importDefault(require("a"));
+a_1.default();

--- a/tests/baselines/reference/exportAssignmentOfExportNamespaceWithDefault.symbols
+++ b/tests/baselines/reference/exportAssignmentOfExportNamespaceWithDefault.symbols
@@ -1,0 +1,39 @@
+=== tests/cases/conformance/externalModules/main.ts ===
+// https://github.com/microsoft/TypeScript/issues/39149
+import a from "a";
+>a : Symbol(a, Decl(main.ts, 1, 6))
+
+a();
+>a : Symbol(a, Decl(main.ts, 1, 6))
+
+=== tests/cases/conformance/externalModules/external.d.ts ===
+declare module "b" {
+>"b" : Symbol("b", Decl(external.d.ts, 0, 0))
+
+    export function a(): void;
+>a : Symbol(a, Decl(external.d.ts, 0, 20), Decl(external.d.ts, 1, 30))
+
+    export namespace a {
+>a : Symbol(a, Decl(external.d.ts, 0, 20), Decl(external.d.ts, 1, 30))
+
+        var _a: typeof a;
+>_a : Symbol(_a, Decl(external.d.ts, 3, 11))
+>a : Symbol(a, Decl(external.d.ts, 0, 20), Decl(external.d.ts, 1, 30))
+
+        export { _a as default };
+>_a : Symbol(_a, Decl(external.d.ts, 3, 11))
+>default : Symbol(default, Decl(external.d.ts, 4, 16))
+    }
+    export default a;
+>a : Symbol(a, Decl(external.d.ts, 0, 20), Decl(external.d.ts, 1, 30))
+}
+
+declare module "a" {
+>"a" : Symbol("a", Decl(external.d.ts, 7, 1))
+
+    import { a } from "b";
+>a : Symbol(a, Decl(external.d.ts, 10, 12))
+
+    export = a;
+>a : Symbol(a, Decl(external.d.ts, 10, 12))
+}

--- a/tests/baselines/reference/exportAssignmentOfExportNamespaceWithDefault.types
+++ b/tests/baselines/reference/exportAssignmentOfExportNamespaceWithDefault.types
@@ -1,0 +1,40 @@
+=== tests/cases/conformance/externalModules/main.ts ===
+// https://github.com/microsoft/TypeScript/issues/39149
+import a from "a";
+>a : typeof import("b").a
+
+a();
+>a() : void
+>a : typeof import("b").a
+
+=== tests/cases/conformance/externalModules/external.d.ts ===
+declare module "b" {
+>"b" : typeof import("b")
+
+    export function a(): void;
+>a : typeof a
+
+    export namespace a {
+>a : typeof a
+
+        var _a: typeof a;
+>_a : typeof a
+>a : typeof a
+
+        export { _a as default };
+>_a : typeof a
+>default : typeof a
+    }
+    export default a;
+>a : typeof a
+}
+
+declare module "a" {
+>"a" : typeof import("a")
+
+    import { a } from "b";
+>a : typeof a
+
+    export = a;
+>a : typeof a
+}

--- a/tests/cases/conformance/externalModules/exportAssignmentOfExportNamespaceWithDefault.ts
+++ b/tests/cases/conformance/externalModules/exportAssignmentOfExportNamespaceWithDefault.ts
@@ -1,0 +1,22 @@
+// @esModuleInterop: true
+// @target: esnext
+// @filename: main.ts
+// @module: commonjs
+// https://github.com/microsoft/TypeScript/issues/39149
+import a from "a";
+a();
+
+// @filename: external.d.ts
+declare module "b" {
+    export function a(): void;
+    export namespace a {
+        var _a: typeof a;
+        export { _a as default };
+    }
+    export default a;
+}
+
+declare module "a" {
+    import { a } from "b";
+    export = a;
+}

